src/analysisd/analysisd.c

```diff
@@ -1661,8 +1661,12 @@ RuleInfo *OS_CheckIfRuleMatch(Eventinfo *lf, RuleNode *curr_node)
     /* If it is a context rule, search for it */
     if(currently_rule->context == 1)
     {
-        if(!currently_rule->event_search(lf, currently_rule))
-            return(NULL);
+
+	if(currently_rule->context_opts & SAME_DODIFF) {
+	} else {
+        	if(!currently_rule->event_search(lf, currently_rule))
+            	return(NULL);
+	}
```

Comment:
Would this not amke more sense writen like this: 

```C
if !(currently_rule->context_opts & SAME_DODIFF) {
     if(!currently_rule->event_search(lf, currently_rule))
     return(NULL);
}
```
 Or am i missing something?